packages/ingest/src/steps/7-publish.ts

```diff
@@ -0,0 +1,260 @@
+import {
+  LessonPlanSchema,
+  type LooseLessonPlan,
+} from "@oakai/aila/src/protocol/schema";
+import type { Prisma, PrismaClientWithAccelerate } from "@oakai/db";
+import { createId } from "@paralleldrive/cuid2";
+import { isTruthy } from "remeda";
+import { z } from "zod";
+
+import { IngestError } from "../IngestError";
+import { loadLessonsAndUpdateState } from "../db-helpers/loadLessonsAndUpdateState";
+import type { Step } from "../db-helpers/step";
+import { getPrevStep } from "../db-helpers/step";
+import type { IngestLogger } from "../types";
+import { chunkAndPromiseAll } from "../utils/chunkAndPromiseAll";
+
+const currentStep: Step = "publishing";
+const prevStep = getPrevStep(currentStep);
+
+/**
+ * Publish ingest lesson_plans and lesson_plan_parts to the rag schema
+ */
+export async function publishToRag({
+  prisma,
+  log,
+  ingestId,
+}: {
+  prisma: PrismaClientWithAccelerate;
+  log: IngestLogger;
+  ingestId: string;
+}) {
+  log.info("Publishing lesson plans and parts to RAG schema");
+
+  const lessons = await loadLessonsAndUpdateState({
+    prisma,
+    ingestId,
+    prevStep,
+    currentStep,
+  });
+
+  log.info(`Loaded ${lessons.length} lessons`);
+
+  /**
+   * Build list of lesson plans to publish
+   */
+  const ragLessonPlans: {
+    oakLessonId?: number;
+    oakLessonSlug: string;
+    ingestLessonId?: string;
+    subjectSlug: string;
+    keyStageSlug: string;
+    lessonPlan: LooseLessonPlan;
+  }[] = [];
+
+  for (const lesson of lessons) {
+    if (!lesson.lessonPlan) {
+      throw new IngestError("Lesson is missing lesson plan", {
+        ingestId,
+        lessonId: lesson.id,
+      });
+    }
+
+    const lessonPlan = LessonPlanSchema.parse(lesson.lessonPlan.data);
+
+    ragLessonPlans.push({
+      oakLessonId: lesson.oakLessonId,
+      oakLessonSlug: lesson.data.lessonSlug,
+      ingestLessonId: lesson.id,
+      subjectSlug: lesson.data.subjectSlug,
+      keyStageSlug: lesson.data.keyStageSlug,
+      lessonPlan,
+    });
+  }
```

Comment:
A functional approach for this (may or may not be better in your eyes!)

```suggestion
  const ragLessonPlans = lessons.map(lesson =>
    if (!lesson.lessonPlan) {
      throw new IngestError("Lesson is missing lesson plan", {
        ingestId,
        lessonId: lesson.id,
      });
    }

    const lessonPlan = LessonPlanSchema.parse(lesson.lessonPlan.data);

    return ({
      oakLessonId: lesson.oakLessonId,
      oakLessonSlug: lesson.data.lessonSlug,
      ingestLessonId: lesson.id,
      subjectSlug: lesson.data.subjectSlug,
      keyStageSlug: lesson.data.keyStageSlug,
      lessonPlan,
    });
  })
```